src/python/pants/base/validation.py

```diff
@@ -24,23 +24,36 @@ def assert_list(obj, expected_type=string_types, can_be_none=True, default=(),
   :param raise_type    : the error to throw if the type is not correct.
   """
   val = obj
+  field_msg = ''
   if val is None:
     if can_be_none:
       val = list(default)
     else:
+      if field_name is None:
+        field_msg="in field '{}':".format(field_name)
+      else:
+        field_msg=''
       raise raise_type(
-        'Expected an object of acceptable type {}, received None and can_be_none is False'
-        .format(allowable))
+        '{} Expected an object of acceptable type {}, received None and can_be_none is False'
```

Comment:
If there is no value for field_msg, then there is going to be an extra space.  The way to fix this is to tack the space onto the end of field_msg after the colon (`:`) when it has a non-blank value.